core/src/test/java/org/elasticsearch/index/query/functionscore/FunctionScoreQueryBuilderTests.java

```diff
@@ -112,66 +90,77 @@ protected FunctionScoreQueryBuilder doCreateTestQueryBuilder() {
-                break;
-            case 2:
-                String script = "5";
-                Map<String, Object> params = Collections.emptyMap();
-                functionBuilder = new ScriptScoreFunctionBuilder(new Script(script, ScriptService.ScriptType.INLINE, MockScriptEngine.NAME, params));
-                break;
-            case 3:
-                RandomScoreFunctionBuilder randomScoreFunctionBuilder = new RandomScoreFunctionBuilder();
-                if (randomBoolean()) {
-                    randomScoreFunctionBuilder.seed(randomLong());
-                } else if (randomBoolean()) {
-                    randomScoreFunctionBuilder.seed(randomInt());
-                } else {
-                    randomScoreFunctionBuilder.seed(randomAsciiOfLengthBetween(1, 10));
-                }
-                functionBuilder = randomScoreFunctionBuilder;
-                break;
-            default:
-                throw new UnsupportedOperationException();
+        ScoreFunctionBuilder<?> functionBuilder;
+        switch (randomIntBetween(0, 3)) {
+        case 0:
+            DecayFunctionBuilder<?> decayFunctionBuilder = createRandomDecayFunction();
+            if (randomBoolean()) {
+                decayFunctionBuilder.setMultiValueMode(randomFrom(MultiValueMode.values()));
+            }
+            functionBuilder = decayFunctionBuilder;
+            break;
+        case 1:
+            FieldValueFactorFunctionBuilder fieldValueFactorFunctionBuilder = fieldValueFactorFunction(fieldValueFactorCompatibleField());
+            if (randomBoolean()) {
+                fieldValueFactorFunctionBuilder.factor(randomFloat());
+            }
+            if (randomBoolean()) {
+                fieldValueFactorFunctionBuilder.missing(randomDouble());
+            }
+            if (randomBoolean()) {
+                fieldValueFactorFunctionBuilder.modifier(randomFrom(FieldValueFactorFunction.Modifier.values()));
+            }
+            functionBuilder = fieldValueFactorFunctionBuilder;
+            break;
+        case 2:
+            String script = "5";
+            Map<String, Object> params = Collections.emptyMap();
+            functionBuilder = new ScriptScoreFunctionBuilder(
+                    new Script(script, ScriptService.ScriptType.INLINE, MockScriptEngine.NAME, params));
+            break;
+        case 3:
+            RandomScoreFunctionBuilder randomScoreFunctionBuilder = new RandomScoreFunctionBuilder();
+            if (randomBoolean()) {
+                randomScoreFunctionBuilder.seed(randomLong());
+            } else if (randomBoolean()) {
+                randomScoreFunctionBuilder.seed(randomInt());
+            } else {
+                randomScoreFunctionBuilder.seed(randomAsciiOfLengthBetween(1, 10));
+            }
+            functionBuilder = randomScoreFunctionBuilder;
+            break;
+        default:
+            throw new UnsupportedOperationException();
```

Comment:
oh I see that you have enabled random score tests....that means that your fix above that I was concerned about did work? sorry it was a long time ago... happy if you fixed it, then you can remove the whole TODO above I guess. (Do run tests with e.g. iters=30 just to make sure)